### PR TITLE
Popper. Strict `PaddingType`, `OffsetType`, `TetherOffsetType` + factories

### PR DESCRIPTION
**Pull request by stefanthaler**
cc @turansky

### CODE REVIEW

File: kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "NOTHING_TO_INLINE",
+)
+package popper.core.modifiers
+
+import kotlinx.js.jso
+import popper.core.Padding
+
+typealias PaddingFunction = (
+    options: OffsetsFunctionOptions,
+) -> PaddingType
```
Comment (turansky) [inline kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt:11]: `Padding`?

File: kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+@file:Suppress(
+    "NOTHING_TO_INLINE",
+)
+package popper.core.modifiers
+
+import kotlinx.js.jso
+import popper.core.Padding
+
+typealias PaddingFunction = (
+    options: OffsetsFunctionOptions,
```
Comment (turansky) [inline kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt:10]: Common name for type required

File: kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+): PaddingType =
+    fn.unsafeCast<PaddingType>()
+
+inline fun Padding(
+    padding: Double
+): PaddingType =
+    padding.unsafeCast<PaddingType>()
+
+inline fun Padding(
+    block: Padding.()->Unit
```
Comment (turansky) [inline kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt:24]: Invalid formatting

File: kotlin-popper/src/main/kotlin/popper/core/modifiers/PreventOverflow.ext.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+@file:Suppress(
+    "NOTHING_TO_INLINE",
+)
+package popper.core.modifiers
+
+inline fun TetherOffset(
+    x: Double,
+): OffsetType =
```
Comment (turansky) [inline kotlin-popper/src/main/kotlin/popper/core/modifiers/PreventOverflow.ext.kt:8]: Invalid result type

File: kotlin-popper/src/main/kotlin/popper/core/modifiers/PreventOverflow.ext.kt [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    x.unsafeCast<OffsetType>()
+
+inline fun TetherOffset(
+    fn: TetherOffsetFunction,
+): TetherOffsetType =
+    fn.unsafeCast<TetherOffsetType>()
+
+typealias TetherOffsetFunction = (
+    options: OffsetsFunctionOptions,
```
Comment (turansky) [inline kotlin-popper/src/main/kotlin/popper/core/modifiers/PreventOverflow.ext.kt:17]: Missed common alias

**Review by turansky — CHANGES_REQUESTED** [review]

File: kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt [hunk 6]
```diff
@@ -0,0 +1,26 @@
+@file:Suppress(
+    "NOTHING_TO_INLINE",
+)
+package popper.core.modifiers
```
Comment (turansky) [inline kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt:4]: Missed new line

**Review by turansky — CHANGES_REQUESTED** [review]

File: kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt (hunk 2 above)
Comment (stefanthaler) [reply kotlin-popper/src/main/kotlin/popper/core/modifiers/Arrow.ext.kt:11]: What about `PositionOptions` ?

File: kotlin-popper/src/main/kotlin/popper/core/modifiers/PreventOverflow.ext.kt (hunk 5 above)
Comment (stefanthaler) [reply kotlin-popper/src/main/kotlin/popper/core/modifiers/PreventOverflow.ext.kt:18]: What for?

**Review by turansky — APPROVED** [review]

**Pull request merged**